engine/table/src/main/java/io/deephaven/engine/table/impl/SourceTable.java

```diff
@@ -363,7 +363,7 @@ protected void destroy() {
     @Override
     @NotNull
     public Stream<PerformanceEntry> sourceEntries() {
-        if (updateSourceRegistrar != null) {
+        if (updateSourceRegistrar != null && locationChangePoller != null) {
```

Comment:
It's better to read the volatile barrier first and make sure you see a fully-constructed `locationChangePoller`.
```suggestion
        if (updateSourceRegistrar != null && locationsInitialized) {
```